static/app/utils/replays/hooks/useReplayData.tsx

```diff
@@ -89,6 +89,7 @@ function useReplayData({
     error: fetchReplayError,
   } = useApiQuery<{data: unknown}>([`/organizations/${orgSlug}/replays/${replayId}/`], {
     staleTime: Infinity,
+    retry: false,
```

Comment:
This is closer to how things worked before https://github.com/getsentry/sentry/pull/63455, so i think it'll be ok to try it out. 

With the retries (default is 3, with a backoff between) it's slow to get that "Replay Not Found" message.